xdsl/transforms/csl_stencil_to_csl_wrapper.py

```diff
@@ -136,6 +136,54 @@ def get_csl_stencil_apply_ops(
                 result.append(apply_op)
         return result
 
+    def _translate_function_args(
+        self, args: Sequence[BlockArgument]
+    ) -> tuple[Sequence[Operation], Sequence[SSAValue]]:
+        """
+        Args of the top-level function act as the interface to the program and need to be translated to writable buffers.
+        """
+        arg_ops: list[Operation] = []
+        arg_op_mapping: list[SSAValue] = []
+        for arg in args:
+            arg_name = arg.name_hint or ("arg" + str(args.index(arg)))
+
+            if isa(arg.type, stencil.FieldType[TensorType[Attribute]]):
+                arg_t = csl_stencil_bufferize.tensor_to_memref_type(
+                    arg.type.get_element_type()
+                )
+                arg_ops.append(
+                    memref.Global.get(
+                        sym_name=StringAttr(arg_name),
+                        sym_type=csl_stencil_bufferize.tensor_to_memref_type(
+                            arg.type.get_element_type()
+                        ),
+                        initial_value=UnitAttr(),
+                        sym_visibility=StringAttr("public"),
+                        # constant: UnitAttr | None = None,
+                        # alignment: int | IntegerAttr[IntegerType] | None = None,
```

Comment:
commented code

Reply:
@dk949 Do we want to specify any of them?

Reply:
I don't believe we have any way to specify alignment in the backend. As for constness, I _think_ these will always be non-const. So you can leave both as default.

Reply:
Thanks, removing commented code.